### PR TITLE
Mayhem

### PR DESCRIPTION
**Pull request by capuanob**
This includes my contributions for harnessing PlutoVG with Mayhem. Please let me know if anything needs to be adjusted!

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -5,9 +5,12 @@ set(CMAKE_C_STANDARD 11)
 project(plutovg C)
 
 add_library(plutovg STATIC)
+target_compile_options(plutovg PUBLIC -fsanitize=fuzzer-no-link)
```
Comment (xansec) [inline CMakeLists.txt:12]: Since you already have built infrastructure under the fuzz directory, I don't see any need to modify the top level CMakeLists.

**Review by xansec — CHANGES_REQUESTED** [review]
Can you comment why you removed the image from this commit?

File: CMakeLists.txt (hunk 1 above)
Comment (capuanob) [reply CMakeLists.txt:12]: I needed to add instrumentation to the library to support fuzzing, which would make sense to do in the root CMakeLists file. I have, in my later submissions, learned to make this optional with a CMake build option and have updated my code to reflect this. Thus, a default build will be unaffected by my changes but specifying -DBUILD_FUZZER=1 will instrument the binary and build the fuzzer directory.

**Comment by capuanob** [comment]
1. Resolved deletion of smiley.png, I believe this was done in error assuming it was an artifact of my testing
2. I think editing the root CMakeList to add the fuzzing subdirectory is inevitable. For build-system clarity, it makes sense to me to have the compiler flags specified here. I have changed it to make this optional though, as making it default was definitely a bad decision. Does that work?

**Comment by xansec** [comment]
Making it option is a better option for sure. LGTM!

**Pull request merged**